### ISSUE TITLE
publish mods on modrinth?

### ISSUE THREAD

**Issue by KraXen72** [maruohon/tweakeroo#252]
hey please can you also publish your mods + maLiLIb on modrinth? it has many advantages over curse forge
https://modrinth.com

thanks

**Comment by maruohon** [comment]
I will probably upload them to Modrinth at some point (but currently the reward program from CurseForge is even my only tiny income...).

However from what I hear apparently at least Tweakeroo would not even be allowed on Modrinth, due to some of its features being rather cheaty, and since the mod is not server-side-opt-in.

**Comment by DragonEggBedrockBreaking** [comment]
Modrinth has a discord server that you can ask questions on btw.

**Comment by DragonEggBedrockBreaking** [comment]
https://discord.gg/gFRbNQ2

**Comment by triphora** [comment]
> However from what I hear apparently at least Tweakeroo would not even be allowed on Modrinth, due to some of its features being rather cheaty, and since the mod is not server-side-opt-in.

I have no reason to believe that is the case. Yes, our rules are stricter than Curse's, but with Tweakeroo specifically, I don't believe that any of the features could meet any of [our criteria for cheats](https://staging.modrinth.com/legal/rules#cheats). I could be wrong (I have not gone through everything with a fine comb), but I don't think that we would reject Tweakeroo.

> but currently the reward program from CurseForge is even my only tiny income...

On an unrelated note, you could likely start a Patreon. Enough people use your mods regularly that I don't think that would be unfeasible :)

**Comment by KraXen72** [comment]
modrinth is planning on giving modders 100% of ad revenue (not implemented yet) so maybe when that gets implemented you could upload them? i understand your concerns. i honetsly feel just like modrinth is wayy easier to use compared to curse, the ui is intuitive and you can see all importatnt info at a glance. if you want to keep them on curseforge tho, its fine, atlauncher supports importing from both.
![image](https://user-images.githubusercontent.com/21956756/126720279-1d6f5873-6904-41c1-b360-ace90973a129.png)

**Comment by Nolij** [comment]
> I will probably upload them to Modrinth at some point (but currently the reward program from CurseForge is even my only tiny income...).
> 
> However from what I hear apparently at least Tweakeroo would not even be allowed on Modrinth, due to some of its features being rather cheaty, and since the mod is not server-side-opt-in.

I see quite a few mods on Modrinth that fit this description (cheaty, no server-side control, etc.), so I don't think that should be an issue.

**Comment by sybenx** [comment]
curseforge is down right now:
![Screenshot Capture - 2022-04-05 - 11-13-09](https://user-images.githubusercontent.com/2048087/161812729-4b139354-b9dc-43b8-abeb-5cf0e0984533.png)
 be nice to have an easy alternative to download on modrinth.

as far as being allowed on modrinth. mods like Flour's Various Tweaks which allow freecam, etc, are already on modrinth.

**Comment by Purpzie** [comment]
> I will probably upload them to Modrinth at some point (but currently the reward program from CurseForge is even my only tiny income...).

It looks like Modrinth's [creator monetization program](https://blog.modrinth.com/p/creator-monetization-beta) has been opened. There's also dedicated settings to put donation links on your mod pages.

**Comment by ghost** [comment]
> > I will probably upload them to Modrinth at some point (but currently the reward program from CurseForge is even my only tiny income...).
> 
> It looks like Modrinth's [creator monetization program](https://blog.modrinth.com/p/creator-monetization-beta) has been opened. There's also dedicated settings to put donation links on your mod pages.

yeah. considering it's now a monetized platform theres no reason to not put it on there platform. mods like tweakermore are on there as well so I dont see why not.

**Comment by triphora** [comment]
If you need help transferring files over and setting up build automation, feel free to drop us a line: support@modrinth.com

**Comment by sylv256** [comment]
i'm baffled as to why this mod (a common cheatmodhack) would be allowed on modrint hconsider in gthat cheatmodhacks are banned

**Comment by ghost** [comment]
> i'm baffled as to why this mod (a common cheatmodhack) would be allowed on modrint hconsider in gthat cheatmodhacks are banned

its not a hack client or any of what you mentioned. lots of these features are clientside (with a few serverside options, which dont work on servers anyway except if mod compatibility is added). if someone does use it for malicious intent it's not very powerful.

**Comment by LunaticWasTaken** [comment]
> its not a hack client or any of what you mentioned. lots of these features are clientside (with a few serverside options, which dont work on servers anyway except if mod compatibility is added). if someone does use it for malicious intent it's not very powerful.

Tweakeroo literally has a Freecam, that allows you to look through walls, be it to find caves, players, or just a simple lava-pool.
If this is not a hack/cheat, then xray isn't either.

I honestly don't understand why the author doesn't add at least a simple serverside opt-out (as it's not really that hard) for the pre-built jars, but hey, this isn't my mod i guess...

**Comment by maruohon** [comment]
I have added it in the development code, but it's not in modern MC version releases yet. More info here: https://github.com/maruohon/tweakeroo/issues/466#issuecomment-1835644832
